### PR TITLE
Ami encryption lambda

### PR DESCRIPTION
**Pull request by sionsmith**
- Added automatic AMI encryption based on the encrypt_ami flag.
- Lambda function kicked off on successful CodeBuild.
- Accepts KMS key arn for EBS encryption

### CODE REVIEW

**Review by mccullya — APPROVED** [review]

File: aws_lambda_function.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
+resource "aws_lambda_function" "ami_encryption_lambda" {
+  count         = "${var.encrypt_ami}"
+  filename      = "${data.archive_file.ami_encryption.0.output_path}"
+  description   = "Responsible for creating AMI with encrypted root volume."
+  function_name = "${var.environment}-AMI-ENCRYPTION-FUNCTION"
```
Comment (richardgutkowski) [inline aws_lambda_function.tf]: Consider putting a `upper()` function around the environment variable as all the rest of the string is in uppercase characters.

File: aws_lambda_iam_policy_document.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "lambda_config_policy" {
+  count         = "${var.encrypt_ami}"
+  statement {
+    actions   = ["ec2:Describe*", "ec2:CreateSnapshot", "ec2:CreateImage", "ec2:CreateTags*", "ec2:CopyImage"]
```
Comment (richardgutkowski) [inline aws_lambda_iam_policy_document.tf:18]: Where is the `effect = "Allow"` statement? is that not required anymore? this is still TF 0.11 right?

**Review by richardgutkowski — APPROVED** [review]
I'm happy to approve, however, do consider my comments from the review. 👍

File: aws_lambda_function.tf (hunk 1 above)
Comment (sionsmith) [reply aws_lambda_function.tf]: added :)

File: aws_lambda_iam_policy_document.tf (hunk 2 above)
Comment (sionsmith) [reply aws_lambda_iam_policy_document.tf:18]: missed that, added it. Still worked without it though

**Pull request merged**